consensus/src/block_storage/block_store.rs

```diff
@@ -166,15 +166,22 @@ impl BlockStore {
     }
 
     async fn try_commit(&self) {
-        // If we fail to commit B_i via state computer and crash, after restart our highest commit cert
-        // will not match the latest commit B_j(j<i) of state computer.
-        // This introduces an inconsistent state if we send out SyncInfo and others try to sync to
-        // B_i and figure out we only have B_j.
-        // Here we commit up to the highest_commit_cert to maintain highest_commit_cert == state_computer.committed_trees.
-        if self.highest_ordered_cert().commit_info().round() > self.ordered_root().round() {
-            let finality_proof = self.highest_ordered_cert().ledger_info().clone();
-            if let Err(e) = self.commit(finality_proof).await {
-                error!(error = ?e, "Commit error during build/rebuild");
+        // reproduce the same batches (important for the commit phase)
+
+        let mut certs = self.inner.read().get_all_quorum_certs_with_commit_info();
+        certs.sort_unstable_by_key(|qc| qc.commit_info().round());
```

Comment:
why need unstable one?

Reply:
this is a bit faster than the stable sort. We do not require the stability here.

Reply:
hmm, maybe if the rounds are different, there is no difference between the two?

Reply:
round should be unique here?

Reply:
yeah in this case the result will be the same. But I think rust stable and unstable sort are actually two different algorithms and the unstable one never allocate extra memory.